### PR TITLE
Private repository usage instructions

### LINKED ISSUE #21: Log in to private npm repository
Hi. Is there a standard way to grant `nix-build` authorization to download from a private npm registry behind a login so that yarn2nix expressions using private npm registries can be consumed?

### PR DESCRIPTION
**Pull request by jkopanski**
Closes #21 

Documents steps required to work with private node packages repository.
I am pretty terrible at nix, so please take a close look at my readme code here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,6 +101,35 @@ Finally, run `nix-build`, and voilà, in `./result/` you find the project with
...
+let
+  auth = { login: "your_login"; taken "your_auth_token"; };
+  config = {
+    packageOverrides = pkgs: rec {
+      fetchurl = otps: pkgs.fetchurl (opts // {
+        netrcPhase = ''
+          cat << EOF > /build/netrc
+            machine repository_url
+              login ${auth.login}
+              password ${auth.token}
```
Comment (Profpatsch) [inline README.md]: This puts the `auth` login/token in the (public) nix store.

File: README.md [hunk 2]
```diff
@@ -101,6 +101,35 @@ Finally, run `nix-build`, and voilà, in `./result/` you find the project with
...
+        '';
+      });
+    };
+  };
+  nixpkgsPath = <nixpkgs>;
+  pkgs = import nixpkgsPath { inherit config };
+  ...
+```
+
+Consider this [wiki](https://nixos.wiki/wiki/Enterprise) article for more help.
```
Comment (Profpatsch) [inline README.md]: Shouldn’t we follow the guide in the wiki more closely here, e.g. the user puts the credentials in his `/etc/netrc` and then just overrides `fetchurl` with `fetchurlBoot`?

File: README.md (hunk 1 above)
Comment (jkopanski) [reply README.md]: Thanks I didn't realized that

File: README.md (hunk 2 above)
Comment (jkopanski) [reply README.md]: probably, will play with this approach in our code and see if it works nicely

Comment (jkopanski) [reply README.md]: the problem with `fetchurlBoot` is that it [requires](https://github.com/NixOS/nixpkgs/blob/ceee97c038c9b0ec48409d3b3313cb77c2543309/pkgs/build-support/fetchurl/boot.nix) `sha256` and npm gives out sha1 and sha512. Do you have any idea haw I could move this forward? @Profpatsch

Comment (Profpatsch) [reply README.md]: Patch it to accept either? `<nix/fetchurl.nix>` can accept all these hashes (`nix-instantiate --eval -E '<nix/fetchurl.nix>'` and open the file name it outputs).

**Comment by Profpatsch** [comment]
I opted to just link to the wiki article for now.

**Pull request merged**